packages/core/src/controllers/verify.ts

```diff
@@ -1,68 +1,140 @@
-import { isBrowser, isNode, isReactNative } from "@walletconnect/utils";
+import {
+  getCryptoKeyFromKeyData,
+  isNode,
+  P256KeyDataType,
+  verifyP256Jwt,
+} from "@walletconnect/utils";
 import { FIVE_SECONDS, ONE_SECOND, toMiliseconds } from "@walletconnect/time";
 
-import { TRUSTED_VERIFY_URLS, VERIFY_CONTEXT, VERIFY_SERVER } from "../constants";
+import { TRUSTED_VERIFY_URLS, VERIFY_CONTEXT, VERIFY_SERVER, VERIFY_SERVER_V2 } from "../constants";
+import { IKeyValueStorage } from "@walletconnect/keyvaluestorage";
 
+type jwk = {
+  publicKey: P256KeyDataType;
+  expiresAt: number;
+};
 export class Verify extends IVerify {
   public name = VERIFY_CONTEXT;
-  private verifyUrl: string;
-  private iframe?: HTMLIFrameElement;
-  private initialized = false;
   private abortController: AbortController;
   private isDevEnv;
-  // the queue is only used during the loading phase of the iframe to ensure all attestations are posted
-  private queue: string[] = [];
-  // flag to disable verify when the iframe fails to load on main & fallback urls.
-  // this means Verify API is not enabled for the current projectId and there's no point in trying to initialize it again.
-  private verifyDisabled = false;
-
-  constructor(public projectId: string, public logger: Logger) {
-    super(projectId, logger);
+  private verifyUrlV2 = VERIFY_SERVER_V2;
+  private publicKey?: jwk;
+
+  constructor(public projectId: string, public logger: Logger, public store: IKeyValueStorage) {
+    super(projectId, logger, store);
     this.logger = generateChildLogger(logger, this.name);
-    this.verifyUrl = VERIFY_SERVER;
     this.abortController = new AbortController();
     this.isDevEnv = isNode() && process.env.IS_VITEST;
+    console.log("Verify v2 init", this.verifyUrlV2);
+    this.init();
   }
 
-  public init: IVerify["init"] = async (params) => {
-    if (this.verifyDisabled) return;
-
-    // ignore on non browser environments
-    if (isReactNative() || !isBrowser()) return;
-
-    const verifyUrl = this.getVerifyUrl(params?.verifyUrl);
-    // if init is called again with a different url, remove the iframe and start over
-    if (this.verifyUrl !== verifyUrl) {
-      this.removeIframe();
-    }
-    this.verifyUrl = verifyUrl;
+  get storeKey(): string {
+    return `verify:public:key`;
+  }
 
-    try {
-      await this.createIframe();
-    } catch (error) {
-      this.logger.info(`Verify iframe failed to load: ${this.verifyUrl}`);
-      this.logger.info(error);
-      // if the iframe fails to load, disable verify
-      this.verifyDisabled = true;
+  public init = async () => {
+    this.publicKey = await this.store.getItem(this.storeKey);
+    console.log("persistedKey", this.publicKey);
+    if (this.publicKey && toMiliseconds(this.publicKey?.expiresAt) < Date.now()) {
+      console.log("public key expired");
+      await this.removePublicKey();
     }
+    if (this.publicKey) return;
+    const key = await this.fetchPublicKey();
```

Comment:
Fetching the key should:
1. Not block SDK initialization, it should run in-parallel and be awaited by the signature verification code
2. Not run unless there is an intention to use the public key. I.e. a pairing was just created on the wallet-side